Subgraph apollo setup

Kenny-Gin1 opened the pull request:
Closes: https://github.com/DXgovernance/davi-monorepo/issues/95 and https://github.com/DXgovernance/davi-monorepo/issues/119

Description: 
Provides an initial setup to support Subgraph in the front end by using the Apollo Client and querying data on multi-chains. 

- Wrapped the app with the `ApolloProvider`
- Implemented the `useGuildConfig` hook
- Dynamically switching networks depending on `chainId`
- Refactored the `wagmi` and the `apollo` clients into their separate folder
- Updated README
- Fixed an issue with the `davi` building with `yarn` instead of `pnpm`

NOTE: There's an `any` used in the `wagmi` client, and it seems to be a deeper issue of symlinking when doing monorepos. Did not find an easy fix for it.

rossneilson commented on apps/davi/src/clients/apollo.ts:
```diff
@@ -0,0 +1,47 @@
+    SUPPORTED_DAVI_NETWORKS.ARBITRUM_TESTNET_ID
+  ),
+  [SUPPORTED_DAVI_NETWORKS.LOCALHOST_ID]: setupApolloClient(
+    SUPPORTED_DAVI_NETWORKS.LOCALHOST_ID
```
Dont we need to add localhost here to test?

rossneilson replied:
Meant to make this comment on line 16

rossneilson commented:
Build error also

Kenny-Gin1 commented:
@rossneilson @dcrescimbeni Please review again. Changes: 
- Added `.graphclient`
- Added localhost URI (think it already worked before since we're also adding it in the `graphclient.yml) but just in case`

The expected behaviour right now is that it can only fetch and make queries to the Subgraph in localhost since that is the only place we have deployed the subgraph.

rossneilson commented on apps/davi/src/stores/modules/SnapshotRepGuild/index.ts:
```diff
@@ -53,7 +54,7 @@ export const snapshotRepGuildImplementation: Readonly<FullGovernanceImplementati
           useVotingResults,
           useVotingPowerOf,
-          useGuildConfig,
+          useGuildConfig: useGuildConfigFromSubgraph,
```
This should be duplicated in non rep guilds also

rossneilson replied:
Also when added manually in there I seem to get 404s in the network tab

Kenny-Gin1 replied:
> Also when added manually in there I seem to get 404s in the network tab

Where does this happen and how to reproduce this?

rossneilson replied:
Fixed with the uri update

rulfo71 commented on apps/davi/src/stores/modules/subgraph/common/useGuildConfig/index.ts:
```diff
@@ -0,0 +1,125 @@
+import { BigNumber } from 'ethers';
```
should we have this on a useGuildConfig.ts and export it from index?

Kenny-Gin1 replied:
Could you elaborate on this? Do you mean that the folder structure should look different?

rulfo71 replied:
yes, i mean having this: 

+ subgraph
|-+ common
| |-+ useGuildConfig 
| | |- index.ts
| | |- useGuildConfig.ts
| | | |- query.graphql

let me know if i'm not being clear

rulfo71 commented:
@Kenny-Gin1 this should be going through useGuildConfigFromSubgraph right? in my case is going through the useGuildConfig from ../common/fetchers

dcrescimbeni commented:
It's working for me now :+1:

Kenny-Gin1 commented:
Great to hear @rossneilson @dcrescimbeni! Thanks for the feedback!

Kenny-Gin1 commented:
> @Kenny-Gin1 this should be going through useGuildConfigFromSubgraph right? in my case is going through the useGuildConfig from ../common/fetchers

This is solved now!
